docs/fastdds/dds_layer/core/waitsets/waitsets.rst

```diff
@@ -65,12 +66,12 @@ For example, if all samples are taken, any ReadCondition associated with the Dat
 triggered before, will see their trigger value changed to false.
 Note that this does not guarantee that WaitSet objects that were separately attached to those
 conditions will not be woken up.
-Once we have trigger_value == true on a condition, it may wake up the attached wait-set.
-The condition transitioning to trigger_value == false does not necessarily ‘unwakeup’ the wait-set,
+Once we have trigger_value == true on a condition, it may wake up the attached :ref:`api_pim_waitset`.
+The condition transitioning to trigger_value == false does not necessarily ‘unwakeup’ the :ref:`api_pim_waitset`,
 as ‘unwakening’ may not be possible in general.
-The consequence is that an application blocked on a wait-set may return from the wait with a list
+The consequence is that an application blocked on a :ref:`api_pim_waitset` may return from the wait with a list
 of conditions, some of which are no longer triggered.
-This is unavoidable if multiple threads are concurrently waiting on separate wait-set objects and
+This is unavoidable if multiple threads are concurrently waiting on separate :ref:`api_pim_waitset` objects and
 taking data associated with the same DataReader entity.
 
```

Comment:
I think we are missing the possibility of the user directly triggering a GuardCondition, for example, to indicate the terminate condition. `GuardCondition::set_trigger_value`